doc/service.md

```diff
@@ -0,0 +1,85 @@
+What is a service?
+==================
+
+A service for Cozy is defined by these criteria:
+
+1. It's some JavaScript code
+2. That is embedded in Cozy-Proxy
+3. As express/americano apps
+4. And exposes a REST API in JSON
+5. to offer functionalities to cozy applications.
+6. This API follows the [JSON api specification](http://jsonapi.org/)
+7. And [Cozy guidelines](https://cozy.github.io/cozy-guidelines/)
+8. With URL prefixed by `/services/:service-name`.
+9. It benefits from Cozy authentication and permissions
+10. And workflows (code review, i18n).
+
+
+Rationale
+---------
+
+### 1. It's some JavaScript code
+
+JS is the lowest common denominator for all the code written at Cozy.
+Technically, a service can be written in something else (CoffeeScript or
+TypeScript) but it has to be compiled simply to JavaScript.
+
+### 2. That is embedded in Cozy-Proxy
```

Comment:
I can understand why host all services in cozy-proxy (it's the only exposed server behind the reverse-proxy, it's already deployed, it permits customization for other parts like Home (you can develop a custom home for your own needs if you want w/o redevelopping all stuff behind the offical one), etc) but I ask myself about the name and architecture of the proxy to do this.

Don't we need to rename it something like "cozy-services", with a smart and clever plugin API to plug services (consider auth as a builtin plugin) to it?